cometd-java/cometd-java-server/cometd-java-server-common/src/main/java/org/cometd/server/BayeuxServerImpl.java

```diff
@@ -1546,4 +1561,85 @@ private static List<String> toChannelList(Object channels) {
 
         return null;
     }
+
+    class Sweeper {
+        private final Logger _logger = LoggerFactory.getLogger(Sweeper.class.getName() + "@" + Integer.toHexString(System.identityHashCode(BayeuxServerImpl.this)));
+
+        private CompletableFuture<Void> asyncSweep() {
+            long before = System.nanoTime();
+            if (_logger.isDebugEnabled()) {
+                _logger.debug("Starting async sweep at {}", Instant.now());
+            }
+
+            CompletableFuture<Void> completableFuture = CompletableFuture.runAsync(BayeuxServerImpl.this::sweepTransports)
```

Comment:
Pass the executor as second parameter so we don't run in the common pool.

Reply:
Done.